sorted_db/db.go

```diff
@@ -92,17 +101,18 @@ func (db *DB) close() {
 		db.f = nil
 	}
 	db.size = -1
+	return
 }
 
 // Remap DB to the same backing file
 func (db *DB) Remap() error {
-	db.RLock()
+	db.mutex.RLock()
```

Comment:
`Remap` calls `Open` after this which will take out it's own `Lock`, so we have to release the `RLock` first